### ISSUE TITLE
Property 'i18n' is missing in type '{ children: Element; }' but required in type 'Readonly<AppProviderProps>'.

### ISSUE THREAD

**Issue by adriancorcoran** [Shopify/quilt#1025]
## Overview

When deploying to the cloud the following error is thrown and the build fails:

```
Property 'i18n' is missing in type '{ children: Element; }' but required in type 'Readonly<AppProviderProps>'.
```

<details><summary>Full output</summary>

```
[at-loader] Checking finished with 1 errors
       9-20T19:20:22Z]
       [client (latest)] Compile failed with errors
       Error: [at-loader] ./app/ui/index.tsx:6:6
       TS2741: Property 'i18n' is missing in type '{ children: Element; }' but required in type 'Readonly<AppProviderProps>'.
       at /tmp/build/node_modules/@shopify/sewing-kit/dist/src/tools/webpack/index.js:104:29
       at Generator.next (<anonymous>)
       at /tmp/build/node_modules/@shopify/sewing-kit/dist/src/tools/webpack/index.js:7:71
       at new Promise (<anonymous>)
```

</details><br>

Checking the [Polaris repo](https://github.com/Shopify/polaris-react#usage), I could see that the translations were not being included in the code in `index.tsx`.

![](https://screenshot.click/20-28-k7gih-mgae8.jpg)

**Fix:**

1. Add `import enTranslations from '@shopify/polaris/locales/en.json';` after `import React from 'react';` in `index.tsx`

1. Add the prop to the `<AppProvider>` like this: `<AppProvider i18n={enTranslations}>`

**Long Term Solution**

I would suggest modifying the code added to the `index.tsx` file during the Quilt installation process to include these two lines.

...

## Consuming repo

I was working on my test repo: https://github.com/Shopify/fluffy-rails-adrian

...

### Scope

- Is this issue related to a specific package?

Not sure

---

## Checklist

- [x] Please delete the labels section before submitting your issue
- [x] I have described this issue in a way that is actionable (if possible)

**Comment by michenly** [comment]
Hi @adriancorcoran, thank you for logging an issue.
However, this is actually an issue for https://github.com/Shopify/polaris-react
and the usage of it is decoupled from `react-i18n` or `i18n` in quilt.

I dont think you should have gotten the type error for not proving `i18n` prop, and I do believe the translation in polaris are included by default. But you may want to ask polaris folks to be sure.

**Comment by michenly** [comment]
cc @BPScott

**Comment by BPScott** [comment]
Heya 👋 

In Polaris v4.0.0 we changed the i18n prop of the AppProvider to be required. This is because in v3 when english was used as a default it meant that any non-english language rendering would still include those default english translations in the bundle, which was wasteful. By requiring an explicit choice of language by the user we avoid including those unneeded translations in non-english bundles.

I'm not sure what instructions or commands @adriancorcoran was running that caused this error but anything that mentions  an `<AppProvider>` should assume Polaris v4 and thus should be updated to include passing the english translations into it [per our readme](https://github.com/shopify/polaris-react#usage).


https://github.com/Shopify/quilt/search?q=AppProvider&unscoped_q=AppProvider mentions a few instructions - https://github.com/Shopify/quilt/blob/f79f52f8b8fd526e8b56755e7207d1d2787970f8/gems/quilt_rails/README.md looks like most relevant one

**Comment by adriancorcoran** [comment]
Hey @BPScott & @michenly :)

To clarify I was following the commands in the Quilt Rails readme on [this page](https://github.com/Shopify/quilt/blob/master/gems/quilt_rails/README.md#example-minimal-reactpolarisquilt-entrypoint). I ran the commands and ran into an error with Typescript needing to be included (possibly due to a template update the issue is opened [here](https://github.com/Shopify/quilt/issues/1024#issuecomment-534545051)). Following that fix (adding in the Typescript dependencies) I hit the error above after deployment to the Shopify Cloudplatform. 

I raised the issue here as the issue occurred without any modification by myself of the commands listed in the [readme](https://github.com/Shopify/quilt/blob/master/gems/quilt_rails/README.md#example-minimal-reactpolarisquilt-entrypoint) (bar fixing the Typescript issue) but also because the Quilt readme is linked from the new [Rails and Node Walkthrough](https://docs.shopifycloud.com/getting_started/rails-with-node-walkthrough) on the Cloudplatform docs page, so it is likely that others may hit this issue in the future and it should be looked at - hope this helps :)
